.migrations/V20250122141100__add_cluster_routing.sql

```diff
@@ -0,0 +1,34 @@
+DO $$
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'cluster_status') THEN
+        CREATE TYPE cluster_status AS ENUM ('active', 'maintenance', 'offline');
+    END IF;
+END$$;
+
+CREATE TABLE IF NOT EXISTS cluster_state (
+    id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    name VARCHAR NOT NULL,
+    cluster_version VARCHAR NOT NULL DEFAULT '',
+    status cluster_status NOT NULL DEFAULT 'active',
+    status_reason VARCHAR,
+    status_since TIMESTAMP WITH TIME ZONE DEFAULT NOW(),
+    capabilities VARCHAR[] NOT NULL DEFAULT '{}',
+    allowed_tiers VARCHAR[] NOT NULL DEFAULT '{}',
```

Comment:
Is this to be used as JSON?

Reply:
Postgres's syntax for arrays is `'{item1,item2,item3}'`,  which would render as `["item1", "item2", "item3"]` in languages like Ruby, Python, etc.

Reply:
See the documentation for more info: https://www.postgresql.org/docs/current/arrays.html#ARRAYS-INPUT